src/datastore/parser.cpp

```diff
@@ -0,0 +1,125 @@
+        std::cout << "---Souvenir:  " << team[28] << std::endl;
+        std::cout << "------Price:  " << team[29] << std::endl;
+        std::cout << "Souvenir ID:  " << team[30] << std::endl;
+        std::cout << "Is deleted?:  " << team[31] << std::endl;
+        std::cout << "---Souvenir:  " << team[32] << std::endl;
+        std::cout << "------Price:  " << team[33] << std::endl;
+        std::cout << std::endl;
+    }
+}
+
+void distanceParseDebug(const std::string & path)
+{
+    std::vector<std::vector<std::string>> distanceData = loadData(path);
+    for (const std::vector<std::string>& stadiumDistance : distanceData)
+    {
+
+        std::cout << "Stadium ID:   " << stadiumDistance[0] << std::endl;
+        std::cout << "Stadium Name: " << stadiumDistance[1] << std::endl;
+        std::cout << "Stadium ID:   " << stadiumDistance[2] << std::endl;
+        std::cout << "Stadium Name: " << stadiumDistance[3] << std::endl;
+        std::cout << "Distance:     " << stadiumDistance[4] << std::endl;
+        std::cout << std::endl;
+    }
+}
+
+std::vector<std::vector<std::string>> loadData(const std::string& path)
+{
+    // Initializing - vector will hold all lines of data within the document
+    std::vector<std::vector<std::string>> allRows;
+    std::ifstream infile(path);
+    int lineCount = 0;
+
+    // Checks if file is open if not throw error
+    if (!infile.is_open()){throw BadFile(QFile::tr(path.c_str()));}
+    std::string line;
+    while (infile.good())
+    {
+        std::getline(infile, line);
+        if (!infile.eof())
+        {
+            if (line[0] == '#' || line[0] == ' '){continue;}
+
+            ++lineCount;
+
+            std::vector<std::string> columns;
+
+            // used for seperating words
+            std::stringstream row(line);
+            std::string word;
+            std::string wordQuotes;
+            int wordCount = 0;
+
+            while (getline(row, word, ','))
+            {
+               if(word[0] == '\"')
+                { 
+                   word.erase(0, 1); //Erase quotation mark
+
+                    std::string extra;
+                    bool finish = false;
+
+                    do
+                    {
+                        std::getline(row, extra, ','); //Extra until a comma
+
+                        if(extra.back() == '\"')
+                        {
+                            extra.erase(extra.size() - 1, 1); //Remove quotation mark at the end
+                            finish = true;
+                        }
+
+                        word += ',' + extra;
+                    } while(!finish);
+                }
+                columns.push_back(word);
+                ++wordCount;
+            }
+            allRows.push_back(columns);
+        }
+    }
+    infile.close();
+    if (lineCount == 0)
+    {
+        throw  BadFileFormat(QString::fromUtf8("*.csv"), QString::fromStdString(path));
```

Comment:
Incorrect exception used. If you want to throw an exception for an empty file, you should be using `BadFile()` and pass something like `Empty file!\n" (file path)`.